### PR TITLE
release: Update image tags for `v0.10.0-rc0`

### PR DESCRIPTION
**Pull request by ckadner**
#### Motivation

Prepare for `v0.10.0` release

#### Modifications

- Update image tags for `v0.10.0-rc0` for `modelmesh`, `modelmesh-runtime-adapter`, `modelmesh-controller`, and `rest-proxy` container images:
   - [x] Edit `newTag` in `config/manager/kustomization.yaml`.
   - [x] Edit the `modelmesh`, `modelmesh-runtime-adapter`, and `rest-proxy` image tags in `config/default/config-defaults.yaml`.
   - [x] Edit the `config/dependencies/quickstart.yaml` file, changing the `kserve/modelmesh-minio-examples` image tag to use the pinned version.
   - [x] Edit the `docs/component-versions.md` file with the version and component versions.
   - [x] Edit the `docs/install/install-script.md` file, updating the `RELEASE` variable in the `Installation` section to the new release branch name.
   - [x] Edit the `docs/quickstart.md` file, updating the `RELEASE` variable in the `Get the latest release` section to the new release branch name.
   - [x] Edit the `scripts/setup_user_namespaces.sh` file, changing the `modelmesh_release` version.
- Update KServe go dependency to `v0.10.0-rc1`

**Note**: the corresponding container images for `modelmesh`, `modelmesh-runtime-adapter` have been created prior to this PR by creating a pre-release on the respective GitHub repositories.

**FYI**: Compare the changes in this PR to the once from the `v0.9.0-rc0` pre-release:
https://github.com/kserve/modelmesh-serving/pull/179/files

### CODE REVIEW

File: config/dependencies/quickstart.yaml [hunk 1]
```diff
@@ -108,7 +108,7 @@ spec:
             - name: MINIO_SECRET_KEY
               value: wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY
           # image: quay.io/cloudservices/minio:latest
-          image: kserve/modelmesh-minio-examples:latest
+          image: kserve/modelmesh-minio-examples:v0.10.0
```
Comment (ckadner) [inline config/dependencies/quickstart.yaml:111]: @njhill -- you might have to push this unless some GitHub action does

**Comment by ckadner** [comment]
@njhill -- please hold off on this until I updated the GitHub actions

File: config/dependencies/quickstart.yaml (hunk 1 above)
Comment (ckadner) [reply config/dependencies/quickstart.yaml:111]: Alright, @yuzisun added me to the [`kserve` org on DockerHub](https://hub.docker.com/orgs/kserve/members) so I was able to push a new tag `v0.10.0` based on the `latest` image

```Bash
docker manifest create kserve/modelmesh-minio-examples:v0.10.0 kserve/modelmesh-minio-examples:latest
docker manifest push kserve/modelmesh-minio-examples:v0.10.0
```

https://hub.docker.com/r/kserve/modelmesh-minio-examples/tags

Comment (ckadner) [reply config/dependencies/quickstart.yaml:111]: thanks @yuzisun 👍🏻

**Comment by ckadner** [comment]
Alright @njhill -- all preconditions should be met for this PR to merge now:

- [x] build + pushed all required `-rc0` container images:
  - [x] kserve/modelmesh:v0.10.0-rc0
  - [x] kserve/modelmesh-runtime-adapter:v0.10.0-rc0
  - [x] kserve/modelmesh-minio-examples:v0.10.0
- [x] GitHub actions are enabled on `release-*` branches

Docker pushes might not work however, as I have not updated the Dockerhub credentials

**Review by njhill — APPROVED** [review]
LGTM

File: go.sum [hunk 2]
```diff
@@ -43,7 +43,7 @@ cloud.google.com/go/compute v1.3.0/go.mod h1:cCZiE1NHEtai4wiufUhW8I8S1JKkAnhnQJW
 cloud.google.com/go/compute v1.5.0/go.mod h1:9SMHyhJlzhlkJqrPAc839t2BZFTSk6Jdj6mkzQJeu0M=
 cloud.google.com/go/compute v1.6.0/go.mod h1:T29tfhtVbq1wvAPo0E3+7vhgmkOYeXjhFvz/FMzPu0s=
 cloud.google.com/go/compute v1.6.1/go.mod h1:g85FgpzFvNULZ+S8AYq87axRKuf2Kh7deLqV/jJ3thU=
-cloud.google.com/go/compute v1.7.0 h1:v/k9Eueb8aAJ0vZuxKMrgm6kPhCLZU9HxFU+AFDs9Uk=
+cloud.google.com/go/compute v1.7.0 h1:v/k9Eueb8aAJ0vZuxKMrg<<<<<<< release-0.10.0-rc0m6kPhCLZU9HxFU+AFDs9Uk=
```
Comment (ckadner) [inline go.sum:46]: Hi @njhill -- looks like a merge conflict artifact?

**Review by ckadner — COMMENTED** [review]
Hi @njhill -- looks like there is a merge conflict artifact left here? Did you merge something into my fork?

File: go.sum (hunk 2 above)
Comment (ckadner) [reply go.sum:46]: oh, I see https://github.com/kserve/modelmesh-serving/pull/298 may have caused conflicts

Comment (ckadner) [reply go.sum]: I rebased this PR on top of #298

**Pull request merged**